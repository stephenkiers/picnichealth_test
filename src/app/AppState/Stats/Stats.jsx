import React, {Component} from 'react'
import PropTypes from 'prop-types'
import ImmutablePropTypes from 'react-immutable-proptypes'


class Stats extends Component {

    render() {
        const {total_records, avg_per_hour, records_last_hour_ratio} = this.props;
        return (
            <div>
                <div className="pill" title="Total">
                    <strong>Total</strong>
                    <span>{total_records}</span>
                </div>
                <div className="pill" title="Avg /hour">
                    <strong>Avg /hour</strong>
                    <span>{avg_per_hour}</span>
                </div>
                <div className={`pill ${records_last_hour_ratio > 0 ? 'positive': 'negative'}`} title="Last hour">
                    <strong>Last hour</strong>
                    <span>{records_last_hour_ratio > 0 ? `+${records_last_hour_ratio}` : records_last_hour_ratio}</span>
                </div>
            </div>
        )
    }
}
Stats.defaultProps = {
};
Stats.propTypes = {
    total_records: PropTypes.number,
    avg_per_hour: PropTypes.number,
    records_last_hour_ratio: PropTypes.number,
}

export default Stats
