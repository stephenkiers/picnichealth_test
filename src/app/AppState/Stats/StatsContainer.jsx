import React, {Component} from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import ImmutablePropTypes from 'react-immutable-proptypes'
import {apiGetStatsUpdate, apiRecordDelete} from "../actions";
import Stats from "./Stats";
import {getStats} from "../../reducers";


class StatsContainer extends Component {
    constructor(props) {
        super(props);
        props.apiGetStatsUpdate();

    }
    componentDidMount() {
        this.interval = setInterval(() => {
            this.props.apiGetStatsUpdate()
        }, 7500);
    }
    componentWillUnmount() {
        if (this.interval) {
            this.interval = clearInterval(this.interval);
        }
    }
    render() {
        return (
            <Stats
                total_records={this.props.stats.get('total_records')}
                avg_per_hour={this.props.stats.get('avg_per_hour')}
                records_last_hour_ratio={this.props.stats.get('records_last_hour_ratio')}
            />
        )
    }
}
StatsContainer.defaultProps = {
};
StatsContainer.propTypes = {
}

const mapStateToProps = (state, ownProps) => ({
    stats: getStats(state)
});

const mapDispatchToProps = (dispatch, ownProps) => ({
    apiGetStatsUpdate(id) {
        dispatch(apiGetStatsUpdate())
    }
});



export default connect(mapStateToProps, mapDispatchToProps)(StatsContainer)
