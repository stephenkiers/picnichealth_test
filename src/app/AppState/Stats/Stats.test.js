import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import Stats from './Stats';

test('Stats displays correct', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(
        <Stats
            total_records={1}
            avg_per_hour={2}
            records_last_hour_ratio={3}
        />
    );
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot();
});