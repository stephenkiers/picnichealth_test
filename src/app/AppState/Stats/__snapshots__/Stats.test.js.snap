// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats displays correct 1`] = `
<div>
  <div
    className="pill"
    title="Total"
  >
    <strong>
      Total
    </strong>
    <span>
      1
    </span>
  </div>
  <div
    className="pill"
    title="Avg /hour"
  >
    <strong>
      Avg /hour
    </strong>
    <span>
      2
    </span>
  </div>
  <div
    className="pill positive"
    title="Last hour"
  >
    <strong>
      Last hour
    </strong>
    <span>
      +3
    </span>
  </div>
</div>
`;
